### PR TITLE
I brought back the staged decoder

### PR DESCRIPTION
**Pull request by cseed**
13.7% improvement (47.3 => 40.8) over and above https://github.com/hail-is/hail/pull/3415.  Tested on a shard of gnomAD with:

```
mt = hl.read_matrix_table('gnomad.mt')
mt._force_count_rows()
```

### CODE REVIEW

File: src/main/scala/is/hail/asm4s/FunctionBuilder.scala [hunk 1]
```diff
@@ -244,7 +244,7 @@ class FunctionBuilder[F >: Null](val parameterTypeInfo: Array[MaybeGenericTypeIn
     try {
       cn.accept(cw)
       bytes = cw.toByteArray
-      CheckClassAdapter.verify(new ClassReader(bytes), false, new PrintWriter(sw1))
+      // CheckClassAdapter.verify(new ClassReader(bytes), false, new PrintWriter(sw1))
```
Comment (cseed) [inline src/main/scala/is/hail/asm4s/FunctionBuilder.scala]: @danking FYI I turned off this checker.  It is INCREDIBLY slow.

File: src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala [hunk 2]
```diff
@@ -108,6 +108,19 @@ class StagedRegionValueBuilder private(val mb: MethodBuilder, val typ: Type, var
 
   def addDouble(v: Code[Double]): Code[Unit] = region.storeDouble(currentOffset, v)
 
+  def allocateBinary(n: Code[Int]): Code[Long] = {
```
Comment (catoverdrive) [inline src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala]: I'm pretty sure that this (and the `emitBinary` case down below) is already handled by `addBinary`, no?

Comment (cseed) [reply src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala]: I don't have the bytes yet so I can't call addBinary, I'm going to get them from the InputStream via readBytes.  readBytes reads directly into the region.

Comment (catoverdrive) [reply src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala]: womp, right.

**Pull request merged**